packages/backend/src/peripherals/ethereum/EthereumClient.ts

```diff
@@ -20,6 +21,19 @@ export class EthereumClient {
     return await this.provider.getLogs(filter)
   }
 
+  async getLogsInRange(blockRange: BlockRange, filter: providers.Filter) {
+    if (blockRange.isEmpty()) {
+      return []
+    }
+    const logs = await this.provider.getLogs({
+      ...filter,
+      fromBlock: blockRange.start,
+      toBlock: blockRange.end - 1,
+    })
+    assert(blockRange.hasAll(logs), 'all logs must be from the block range')
+    return logs
+  }
```

Comment:
I like this change a lot, but I'm curious about the rename from `from` and `to` to `start` and `end`? Does `end` communicate exclusivity better?

Reply:
> It would be just too convinient to pass from and to into Filter's fromBlock and toBlock, so I changed the naming so that maybe some warning light would flash in the programmer's mind

👍 Makes total sense.